cmd/subnetcmd/removeValidator.go

```diff
@@ -0,0 +1,237 @@
+		useLedger = true
+		if keyName != "" {
+			return ErrStoredKeyOnMainnet
+		}
+	default:
+		return errors.New("unsupported network")
+	}
+
+	// used in E2E to simulate public network execution paths on a local network
+	if os.Getenv(constants.SimulatePublicNetwork) != "" {
+		network = models.Local
+	}
+
+	sc, err := app.LoadSidecar(subnetName)
+	if err != nil {
+		return err
+	}
+
+	subnetID := sc.Networks[network.String()].SubnetID
+	if subnetID == ids.Empty {
+		return errNoSubnetID
+	}
+
+	controlKeys, threshold, err := subnet.GetOwners(network, subnetID)
+	if err != nil {
+		return err
+	}
+
+	// get keys for add validator tx signing
+	if subnetAuthKeys != nil {
+		if err := prompts.CheckSubnetAuthKeys(subnetAuthKeys, controlKeys, threshold); err != nil {
+			return err
+		}
+	} else {
+		subnetAuthKeys, err = prompts.GetSubnetAuthKeys(app.Prompt, controlKeys, threshold)
+		if err != nil {
+			return err
+		}
+	}
+	ux.Logger.PrintToUser("Your subnet auth keys for add validator tx creation: %s", subnetAuthKeys)
+
+	if nodeIDStr == "" {
+		nodeID, err = promptNodeID()
+		if err != nil {
+			return err
+		}
+	} else {
+		nodeID, err = ids.NodeIDFromString(nodeIDStr)
+		if err != nil {
+			return err
+		}
+	}
+
+	// TODO check that this guy actually is a validator on the subnet
+
+	ux.Logger.PrintToUser("NodeID: %s", nodeID.String())
+	ux.Logger.PrintToUser("Network: %s", network.String())
+	ux.Logger.PrintToUser("Inputs complete, issuing transaction to remove the sepecified validator information...")
+
+	// get keychain accesor
+	kc, err := GetKeychain(useLedger, ledgerAddresses, keyName, network)
+	if err != nil {
+		return err
+	}
+	deployer := subnet.NewPublicDeployer(app, useLedger, kc, network)
+	isFullySigned, tx, err := deployer.RemoveValidator(subnetAuthKeys, subnetID, nodeID)
+	if err != nil {
+		return err
+	}
+	if !isFullySigned {
+		if err := SaveNotFullySignedTx(
+			"Remove Validator",
+			tx,
+			network,
+			subnetName,
+			subnetID,
+			subnetAuthKeys,
+			outputTxPath,
+			false,
+		); err != nil {
+			return err
+		}
+	}
+
+	return err
+}
+
+func removeFromLocal(subnetName string) error {
+	sc, err := app.LoadSidecar(subnetName)
+	if err != nil {
+		return err
+	}
+
+	subnetID := sc.Networks[models.Local.String()].SubnetID
+	if subnetID == ids.Empty {
+		return errNoSubnetID
+	}
+
+	// Get NodeIDs of all validators on the subnet
+	validators, err := subnet.GetSubnetValidators(subnetID)
+	if err != nil {
+		return err
+	}
+
+	// construct list of validators to choose from
+	validatorList := make([]string, len(validators))
+	for i, v := range validators {
+		validatorList[i] = v.NodeID.String()
+	}
+
+	// Get NodeID of the node we want to remove
+	nodeIDStr, err := app.Prompt.CaptureList("Choose a validator to remove", validatorList)
+	if err != nil {
+		return err
+	}
+
+	// Convert NodeID string to NodeID type
+	nodeID, err := ids.NodeIDFromString(nodeIDStr)
+	if err != nil {
+		return err
+	}
+
+	testKey := genesis.EWOQKey
```

Comment:
this assumes from ANR key. probably we want to add remove validator support to ANR abstracting its wallet

Reply:
https://github.com/ava-labs/avalanche-cli/issues/738